### PR TITLE
Build GHCR images

### PR DESCRIPTION
**Pull request by Xiangs18**

### CODE REVIEW

File: .github/workflows/docker-publish.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+on:
+  workflow_dispatch:
+  push:
+    branches: [ "main", "master", "develop" ]
+    # Publish semver tags as releases.
+    tags:
+      - 'v[0-9]+.[0-9]+.[0-9]+'
+      - '[0-9]+.[0-9]+.[0-9]+'
+      - '[0-9]+.[0-9]+.[0-9]+-*'
```
Comment (MrCreosote) [inline .github/workflows/docker-publish.yml:11]: Hmm, not sure what'll happen when we push a tag and also create a release on that tag. We may need to update this

**Review by MrCreosote — APPROVED** [review]

File: .github/workflows/docker-publish.yml (hunk 1 above)
Comment (Xiangs18) [reply .github/workflows/docker-publish.yml:11]: Why do you believe this could be a concern? Could you provide a more detailed explanation?

Comment (MrCreosote) [reply .github/workflows/docker-publish.yml:11]: We push tag 1.2.3 -> image build with tag 1.2.3
We make a release from that build -> image build with tag 1.2.3

Comment (Xiangs18) [reply .github/workflows/docker-publish.yml:11]: I assume they will be shown under `Recent tagged image versions`. Why is this a concern?

Comment (MrCreosote) [reply .github/workflows/docker-publish.yml:11]: I didn't say it was a concern, I said I'm not sure what'll happen and we should keep an eye on it

**Pull request merged**